sdk/monitor/azure-monitor-query/src/main/java/module-info.java

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/**
+ * Module definition for azure-monitor-query.
+ */
+module com.azure.monitor.query {
+    requires transitive com.azure.core;
+    requires transitive com.azure.core.experimental;
+    exports com.azure.monitor.query;
+    exports com.azure.monitor.query.models;
+    opens com.azure.monitor.query.metrics.implementation.models to com.fasterxml.jackson.databind;
+    opens com.azure.monitor.query.log.implementation.models to com.fasterxml.jackson.databind;
```

Comment:
I initially saw some of our libraries needing opens to azure core too. My conclusion was that we need those since  - the service interface in our impl classes were not public, which we have now changed.

Does that mean we can remove opens for the core libraries from them?

Reply:
It depends on the library. Generally, if you can run samples and tests without opening packages to com.azure.core, then it is not required. If samples don't work, then it might be required.